### DISCUSSION TITLE
Getting 403 error on Livewire resource on shared hosting

### DISCUSSION THREAD

**Discussion by JakubJV** [livewire/livewire#8000] [1. Help] [answered]
"Hello everyone, I'm encountering a 403 error on shared hosting after deploying my web app, which is built using Livewire v3 and Laravel 10. I didn't published livewire config according to docs.

![307244271-1167f494-8b62-4fae-aed8-1d69f286b535](https://github.com/livewire/livewire/assets/102606824/9a07f119-ce69-441e-9072-7fc224e8400f)
![307244273-ca1d3f0c-abc8-4755-a92f-e80e7aff5ed8](https://github.com/livewire/livewire/assets/102606824/69abeb69-f451-4655-b638-40b249ffe0df)
![307244276-50e0af2d-b006-43e8-ae21-b862a755a69a](https://github.com/livewire/livewire/assets/102606824/cbcf6816-2e00-4ec0-94cb-6d471b5c3536)

This is my routes -> 
```
<?php

use Livewire\Drawer\Utils;
use App\Livewire\Admin\Users;
use App\Livewire\Forms\Login;
use App\Livewire\Admin\Profile;
use App\Livewire\Forms\Reserve;
use App\Livewire\Forms\Register;
use App\Livewire\Forms\BlacklistForm;
use App\Livewire\Forms\BusinessHours;
use Illuminate\Support\Facades\Route;



Route::get('/', function () {
    return view('app');
});


Route::get('/login', Login::class)->name('login');
Route::get('/register', Register::class)->name('register')->middleware('auth');
Route::get('/business-hours', BusinessHours::class)->middleware('auth')->name('businessHours');
Route::get('/reservations', Reserve::class)->middleware('auth')->name('reservations');
Route::get('/users', Users::class)->middleware('auth')->name('users');
Route::get('/profile', Profile::class)->middleware('auth')->name('profile');
Route::get('/blacklist', BlacklistForm::class)->middleware('auth')->name('blacklist');
```

and this is how I'm using directives in layouts
```
    @livewireStyles
    <script id="Cookiebot" src="https://consent.cookiebot.com/uc.js" data-cbid="e8cc7f0c-2918-4642-a3f9-87e4fdb602fd"  type="text/javascript"></script>
</head>

<body>
    <header>
        <livewire:navbar />
    </header>
    <main>
        <livewire:home />
        <livewire:about />
        <livewire:services />
        <livewire:booking>
    </main>

    <footer>
        <livewire:footer />
    </footer>
    <a href="#úvod" class="to-top" id="top-arrow">
        <svg xmlns="http://www.w3.org/2000/svg" width="42" height="42" fill="currentColor"
            class="bi bi-arrow-up-square-fill" viewBox="0 0 16 16">
            <path
                d="M2 16a2 2 0 0 1-2-2V2a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2zm6.5-4.5V5.707l2.146 2.147a.5.5 0 0 0 .708-.708l-3-3a.5.5 0 0 0-.708 0l-3 3a.5.5 0 1 0 .708.708L7.5 5.707V11.5a.5.5 0 0 0 1 0" />
        </svg>
    </a>
    @livewireScripts
    <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/js/bootstrap.bundle.min.js"
        integrity="sha384-C6RzsynM9kWDrMNeT87bh95OGNyZPhcTNXj1NW7RuBCsyN/o0jlpcV8Qyq46cDfL" crossorigin="anonymous" defer>
    </script>
    <script src="{{ url('/js/app.js') }}" defer></script>

    <script src="https://cdn.jsdelivr.net/npm/sweetalert2@11" defer></script>

    <x-livewire-alert::scripts />

</body>
```

**Comment by shubdragon** [comment]
can you show .htaccess file on root folder, if not the create a new .htaccess file and  add this line (**not the .htaccess present on public folder)
`    <IfModule mod_rewrite.c>
     RewriteEngine On
     RewriteRule ^(.*)$ public/$1 [L]
   </IfModule>
`

**Reply by JakubJV** [reply]
@shubdragon  this is my htaccess file , i will try update it like you are saying and let you know, thanks for advice :) 
```
<IfModule mod_rewrite.c>
    <IfModule mod_negotiation.c>
        Options -MultiViews -Indexes
    </IfModule>

    RewriteEngine On

    # Handle Authorization Header
    RewriteCond %{HTTP:Authorization} .
    RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]

    # Redirect Trailing Slashes If Not A Folder...
    RewriteCond %{REQUEST_FILENAME} !-d
    RewriteCond %{REQUEST_URI} (.+)/$
    RewriteRule ^ %1 [L,R=301]

    # Send Requests To Front Controller...
    RewriteCond %{REQUEST_FILENAME} !-d
    RewriteCond %{REQUEST_FILENAME} !-f
    RewriteRule ^ index.php [L]
</IfModule>
```

**Reply by shubdragon** [reply]
sure, but  it is looks like the .htacces you are showing from from public folder you need to add a new .htaccess on your root folder with other files e.g `.env` and point your domain to laravel root where all the file.env is located

**Reply by JakubJV** [reply]
@shubdragon  ahhh, now I understand perfectly, okay, will try this aswell and let you know! :)

**Reply by JakubJV** [reply]
I have got it like this,  in www folder I have content of public folder, and in barbershop27 folder I've got root folder with .env, livewire components..., created new htaccess file in root directory in folder barbershop, I really don't know if I did it correctly but still getting 403 on livewire script :/ @shubdragon 

![www folder](https://github.com/livewire/livewire/assets/102606824/e1192dfa-8573-40c3-9430-08cdac6a6c0a)
![htaccess 152239](https://github.com/livewire/livewire/assets/102606824/a8069e17-e323-4c07-bccb-a714cc266ef9)
![root folder](https://github.com/livewire/livewire/assets/102606824/a23d1b9a-320c-4f7a-ac90-1acb8eaae1a8)
![server folder](https://github.com/livewire/livewire/assets/102606824/92fcc8e2-4524-4362-a21c-abba596c2e18)

**Comment by MasterAdmin2** [comment]
i have the solution to this problem, if u wanna know I will give you a solution, just ping me at **@sarthakgarg0912** -> discord

**Reply by JakubJV** [reply]
@MasterAdmin2  can't add you on discord :/

**Reply by MasterAdmin2** [reply]
Add me as a friend on Discord! Invite expires in 1 week: https://discord.com/invite/cMbWffkm

**Reply by joshhanley** [reply]
@MasterAdmin2 if you have a solution, could you please post it here instead of on Discord so everyone benefits from it?

That way if someone else has this problem in future they can easily find an answer 🙂

Thanks!

**Reply by JakubJV** [reply]
@joshhanley  so really i don't need to publish config in Livewire 3 ?

**Reply by joshhanley** [reply]
@JakubJV nope, should work out of the box. The 403 forbidden suggests that there is a permissions issue within your app or with your hosting provider. I'm guessing everything is working fine locally? If so then that suggest something wrong with the hosting setup.

**Reply by JakubJV** [reply]
@joshhanley so i dont need edit any asset url or something like that, ok so i will try to solve this with hosting provider

**Reply by joshhanley** [reply]
@JakubJV generally not. Every set up can be different though. If you were getting a 404 error then I would say then you would need to adjust the asset url. But for you, the asset url is resolving fine but returning a 403 which suggests a permission problem.

**Reply by JakubJV** [reply]
Yeah, I'll try to solve this with hosting provider, but definetly thank you for your time! :)

**Reply by JakubJV** [reply]
@joshhanley have you idea what is this ? i thought that livewire doesnt use csrf directives in forms :/
![Snímek obrazovky 2024-02-28 105603](https://github.com/livewire/livewire/assets/102606824/cc045e2f-a58b-46a8-94a8-18b164d18f3e)

**Reply by PhiloNL** [reply]
> @joshhanley have you idea what is this ? i thought that livewire doesnt use csrf directives in forms :/
> 
> ![Snímek obrazovky 2024-02-28 105603](https://github.com/livewire/livewire/assets/102606824/cc045e2f-a58b-46a8-94a8-18b164d18f3e)
> 

Can you check your code and see if you have any unclosed elements like divs, spans, buttons, etc? This often leads to this error.

**Reply by JakubJV** [reply]
@PhiloNL  it was unclosed div, thanks! :)

**Comment by MasterAdmin2** [comment]
> @MasterAdmin2 if you have a solution, could you please post it here instead of on Discord so everyone benefits from it?
> 
> That way if someone else has this problem in future they can easily find an answer 🙂
> 
> Thanks!


i can explain you in discord, you can post it by yourself. thanks

**Comment by JakubJV** [comment, marked as answer]
Problem was somewhere on hosting, I don't really know wthat was that but they updated some permissions and it works now.

**Reply by joshhanley** [reply]
@JakubJV great to hear you got it sorted 🙂

**Comment by hasan-aftab** [comment]
https://stackoverflow.com/a/77287647/22098128

This worked for me
